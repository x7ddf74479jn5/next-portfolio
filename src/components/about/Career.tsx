import { Timeline, timelines } from "src/contents/timelines";
import styles from "src/styles/components/about/Career.module.scss";
import { WrapInSection } from "../../layouts/Section";

const Career: React.VFC = () => {
  return (
    <WrapInSection id={"career"}>
      <h2 className={styles.headlineAbout}>Career</h2>
      <ul className={styles.timeline}>
        {timelines.map((timeline, index) => {
          return <TimelineItem timeline={timeline} key={index} />;
        })}
      </ul>
    </WrapInSection>
  );
};

type TimeLineItemProps = {
  timeline: Timeline;
};

const TimelineItem: React.VFC<TimeLineItemProps> = ({ timeline }) => {
  return (
    <li>
      <time dateTime={timeline.datetime}>{timeline.datetime}</time>
      <div className={styles.timelineIcon}></div>
      <div className={styles.timelineBox}>
        <h3> {timeline.title} </h3>
        <p> {timeline.description} </p>
      </div>
    </li>
  );
};

export default Career;
