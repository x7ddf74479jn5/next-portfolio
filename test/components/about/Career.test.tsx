import { render } from "@testing-library/react";
import Career from "src/components/about/Career";

describe("Career", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Career />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("shows headline and has id", () => {
    const { container } = render(<Career />);
    const targetElement = container.getElementsByClassName("section")[0];
    expect(container).toHaveTextContent("Career");
    expect(targetElement).toHaveAttribute("id", "career");
  });

  it("timeline list has over one item", () => {
    const { container } = render(<Career />);
    const targetElement = container.getElementsByClassName("timeline");
    expect(targetElement.length).toBeGreaterThan(0);
  });
});
