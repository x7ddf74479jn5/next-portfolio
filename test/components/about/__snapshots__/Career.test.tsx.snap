// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Career matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="section"
    id="career"
  >
    <div
      class="sectionWrapIn"
    >
      <h2
        class="headlineAbout"
      >
        Career
      </h2>
      <ul
        class="timeline"
      >
        <li>
          <time
            datetime="2012-10"
          >
            2012-10
          </time>
          <div
            class="timelineIcon"
          />
          <div
            class="timelineBox"
          >
            <h3>
               新卒で大手外食チェーン店に入社 
            </h3>
            <p>
               店長候補として入社し、都内拠点で店舗オペレーションを学んでいました。従業員マネジメントに力を入れ、アルバイトの離職率改善へ取り組みました。 
            </p>
          </div>
        </li>
        <li>
          <time
            datetime="2016-04"
          >
            2016-04
          </time>
          <div
            class="timelineIcon"
          />
          <div
            class="timelineBox"
          >
            <h3>
               地元に戻り、喫茶店の調理職へ 
            </h3>
            <p>
               地元食材を用いた付加価値の高い商品を提供しています。自分が独学したもののもうひとつが料理で、それが嵩じて調理職に。 
            </p>
          </div>
        </li>
        <li>
          <time
            datetime="2016-07"
          >
            2016-07
          </time>
          <div
            class="timelineIcon"
          />
          <div
            class="timelineBox"
          >
            <h3>
               プログラミングを独学 
            </h3>
            <p>
               以前よりクラウドソーシングでWebライティングの経験があり、よりスキルアップするためプログラミングの勉強を始めました。Javascript &gt; jQuery &gt; PHP &gt; WordPress &gt; ... 
            </p>
          </div>
        </li>
        <li>
          <time
            datetime="2017-01"
          >
            2017-01
          </time>
          <div
            class="timelineIcon"
          />
          <div
            class="timelineBox"
          >
            <h3>
               Web制作の仕事を開始 
            </h3>
            <p>
               クラウドソーシングを利用して案件を受注できるようになり、本格的にWeb制作の仕事を開始。 
            </p>
          </div>
        </li>
        <li>
          <time
            datetime="Now"
          >
            Now
          </time>
          <div
            class="timelineIcon"
          />
          <div
            class="timelineBox"
          >
            <h3>
               現在 
            </h3>
            <p>
               React及びモダンフロントエンド開発のエコシステム、応用情報技術者試験などの学習を継続しながら転職活動を進めています。 
            </p>
          </div>
        </li>
      </ul>
    </div>
  </section>
</DocumentFragment>
`;
